lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,327 @@
+          size: 24,
+          color: selectedUserIds.isEmpty
+            ? designVariables.icon.withFadedAlpha(0.5)
+            : designVariables.icon)]));
+}
+
+class NewDmSearchBar extends StatelessWidget {
+  const NewDmSearchBar({
+    super.key,
+    required this.searchController,
+    required this.scrollController,
+    required this.selectedUserIds});
+
+  final TextEditingController searchController;
+  final ScrollController scrollController;
+  final Set<int> selectedUserIds;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    return Container(
+      width: MediaQuery.of(context).size.width,
+      constraints: const BoxConstraints(
+        minHeight: 44,
+        maxHeight: 124),
+      padding: const EdgeInsets.symmetric(horizontal: 14,vertical:11),
+      decoration: BoxDecoration(color: designVariables.bgSearchInput),
+      child: SingleChildScrollView(
+        controller: scrollController,
+        child: Row(children: [
+          Expanded(child: Wrap(
+            spacing: 6,
+            runSpacing: 4,
+            crossAxisAlignment: WrapCrossAlignment.center,
+            children: [
+              ...selectedUserIds.map((userId) => SelectedUserChip(userId: userId)),
+              _buildSearchField(designVariables, zulipLocalizations)]))])));
+  }
+
+  Widget _buildSearchField(DesignVariables designVariables,
+      ZulipLocalizations zulipLocalizations) => IntrinsicWidth(
+        child: SizedBox(
+          height: 22,
+          child: TextField(
+            controller: searchController,
+            cursorColor: designVariables.foreground,
+            style: const TextStyle(
+              fontSize: 17,
+              height: 1.0,
+            ),
+            scrollPadding: EdgeInsets.zero,
+            decoration: InputDecoration(
+              hintText: zulipLocalizations.newDmSheetSearchHint,
+              hintStyle: TextStyle(
+                fontSize: 17,
+                height: 1.0,
+                color: designVariables.textInput.withFadedAlpha(0.5)),
+              border: InputBorder.none,
+            ))));
+}
+
+class SelectedUserChip extends StatelessWidget {
+  const SelectedUserChip({
+    super.key,
+    required this.userId});
+
+  final int userId;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final user = store.users[userId]!;
+
+    return Container(
+      height: 22,
+      decoration: BoxDecoration(color: designVariables.bgMenuButtonSelected, borderRadius: BorderRadius.circular(3)),
+      child: Row(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Avatar(userId: userId, size: 22, borderRadius: 3),
+          Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 5, vertical: 3),
```

Comment:
The end inset is not 5px:

![image](https://github.com/user-attachments/assets/d020ab6c-c4a1-4714-803b-af4068c2e383)

```suggestion
            padding: EdgeInsetsDirectional.fromSTEB(5, 3, 4, 3),
```

We use `EdgeInsetsDirectional` because the start/end insets are different. This is for RTL where left and right are flipped.